python/cudf/dataframe/dataframe.py

```diff
@@ -1368,10 +1383,21 @@ def merge(self, right, on=None, how='left', lsuffix='_x', rsuffix='_y',
              1    2 12.0   11.0
              4    3 13.0
              2    4 14.0   12.0
-
         """
         _gdf.nvtx_range_push("CUDF_JOIN", "blue")
 
+        if right_on:
+            on = right_on
+            self[right_on] = self.index
+        elif left_on:
+            on = left_on
+            right[left_on] = right.index
+
+        if left_index and right_index:
+            on = self.LEFT_RIGHT_INDEX_NAME
+            self[on] = self.index
+            right[on] = right.index
```

Comment:
I have to admit that the logic above confuses me.

Reply:
`left_index = right_index = True` as arguments means to perform the merge using only the indexes of each df. Our join system joins based on columns and ignores indexes, so I put the indexes in as columns and give them a unique column name. It appears to work, matching pandas in examples of varying complexity. I cannot vouch for 100% pandas compatibility though admittedly.